onnx/src/androidMain/kotlin/org/jetbrains/kotlinx/dl/api/inference/onnx/objectdetection/SSDLikeModel.kt

```diff
@@ -24,18 +27,33 @@ public class SSDLikeModel(override val internalModel: OnnxInferenceModel, metada
 
     override val classLabels: Map<Int, String> = Coco.V2017.labels(zeroIndexed = true)
 
-    override var targetRotation: Float = 0f
+    override var targetRotation: Int = 0
 
     override val preprocessing: Operation<Bitmap, Pair<FloatArray, TensorShape>>
         get() = pipeline<Bitmap>()
             .resize {
                 outputHeight = internalModel.inputDimensions[0].toInt()
                 outputWidth = internalModel.inputDimensions[1].toInt()
             }
-            .rotate { degrees = targetRotation }
+            .rotate { degrees = targetRotation.toFloat() }
             .toFloatArray { layout = TensorLayout.NHWC }
 
     override fun close() {
         internalModel.close()
     }
 }
+
+/**
+ * Returns the detected object for the given image sorted by the score.
+ * Internal preprocessing is updated to rotate image to match target orientation.
+ * After prediction, internal preprocessing is restored to the original state.
+ *
+ * @param [imageProxy] Input image.
+ * @param [topK] The number of the detected objects with the highest score to be returned.
+ * @return List of [DetectedObject] sorted by score.
+ */
+public fun SSDLikeModel.detectObjects(imageProxy: ImageProxy, topK: Int = 3): List<DetectedObject> {
```

Comment:
We can write something like this to enable this method for all object detection models:

```kotlin
public fun ObjectDetectionModelBase<Bitmap>.detectObjects(imageProxy: ImageProxy, topK: Int = 3): List<DetectedObject> {
    if (this is CameraXCompatibleModel) {
        val currentRotation = targetRotation
        targetRotation = imageProxy.imageInfo.rotationDegrees
        return detectObjects(imageProxy.toBitmap(), topK).also { targetRotation = currentRotation }
    }
    return detectObjects(imageProxy.toBitmap(applyRotation = true), topK)
}
```
Not pretty, but gets the job done. What do you think?